### ISSUE TITLE
TLE for Lab4

### ISSUE THREAD

**Issue by beefsausagee** [nus-cs2030/2324-s1#91]
# Summary

Description of the question / problem here.
for testcase 11/18 i kept getting Time limit exceed. My TA said its due to infinite loop. Did anyone encounter this issue too?

## Description:

Clarifications or details with respect to the issue you are facing

## Code Snippets

Attach code snippets (if any)

## Screenshots

Screenshots (if any)
![Uploading Screenshot 2023-11-08 at 7.02.03 PM.png…]()

**Comment by leeshinn** [comment]
you could check your simulator class, maybe you're failing to remove events from your queue?

**Comment by Yihan2407** [comment]
Yup TLE is most likely due to infinite loops occurring in the case of CS2030, as we are -- to my knowledge -- not typically assessed on how efficient our code is.

I think the most common source of infinite loops occurring in the Discrete Event Simulator is the implementation of the server queue, be it the human queue or self-checkout queue, so that is where you would most likely want to look at first to try and resolve this. Make sure that your code is able to work even when there are multiple people in the queue, as that is likely where the infinite loop is occurring, i.e., events not being resolved properly.

**Comment by ChuaLonghorn** [comment]
I'm also having a similar issue. What is the general concept that you are using to resolve this TLE through implementing the checkout queues (human & self-checkout) to ensure that events are resolved properly?

**Comment by ChoonHean** [comment]
I ran into this issue, with the next test case(9_12) as it had more than 100 customers. The issue with my code was that that test case had so many customers, that it would be possible for a queue to build and empty before the next customer arrives, and since there is no queue, the customer could then be immediately served without queueing. Maybe check for that. Also, the first ~10 test cases only test for rest and not self checkout, so check for your self checkout implementation.

**Comment by roderickvo** [comment]
I had this issue previously because I realized that I did not update my server list accordingly in one of my event classes

**Comment by vvhuiling** [comment]
may check the timestamp of your wait event. I had similar issue before because of the mismatch of the serviceTime supplier. remember all customers share the same supplier, so if one get serviceTime with wrong order, the rest will also get wrong ones, leading to the mismatch timestamp and infinite looping.

**Comment by yentheng0110** [comment]
Hello! I encountered a similar issue previously, the problem lied in my Wait class, I returned the Wait event start time as the time stamp of the next Wait event and hence, infinite loops occurred.
